src/common/snapshots.rs

```diff
@@ -43,13 +47,17 @@ pub async fn delete_shard_snapshot(
     shard_id: ShardId,
     snapshot_name: String,
 ) -> Result<(), StorageError> {
+    // This future is cancel-safe
+    //
+    // TODO: Explain why?
```

Comment:
I assume because there's only a single 'essential' thing happening: remove_file. That either happens or not, and won't leave some broken state.

But, can we cancel the future in the middle of this file operation? And if so, would that leave some broken state?

Reply:
Have you ever seen "partially deleted files"? (:

There are still only two options: either the file is deleted, or not.

Reply:
I've been thinking too much about futures here :dizzy_face: :dizzy: